### ISSUE TITLE
Erroneous compression algorithm info

### ISSUE THREAD

**Issue by rask** [humbug/box#20]
Attempted to follow installation instructions for a project-installation with `bamarni/composer-bin-plugin` as such:

```sh
$ composer require --dev bamarni/composer-bin-plugin
$ composer bin box require --dev humbug/box
# then inserted required commands to `post-install-cmd` and `post-update-cmd`
$ composer update # just in case
```

When attempting to run

    $ vendor/bin/box help build

I receive the following error:

    PHP Fatal error: Uncaught error: Class 'KevinGH\Box\Application' not found in /path/to/project/vendor-bin/box/vendor/humbug/box/bin/box:19

It seems like `bamarni/composer-bin-plugin` is not running `composer install` on packages properly as the `humbug/box` vendor directory is missing a `vendor` directory.

I will try and see if a global installation changes anything, but I would like to carry the dep as a local dep with the project to make CI build pipelines just a little simpler to set up.

**Comment by theofidry** [comment]
Try to do:

```
composer bin box require --dev humbug/box:dev-master
```

This fork is still not very stable yet. The tests are green so it should be ok but I prefer to be transparent and say that I it's not ready for an alpha or beta release yet :)

Also regarding bamarni you should see when it executes the install, but if you want to run an install/update or anything again, you should do:

```
composer bin box update
# or
composer bin all
```

Otherwise you're running the composer command for the root `composer.json`,  not the `vendor-bin/box/composer.json` one

**Comment by rask** [comment]
I'll see what the dev-master version says, thanks. :)

Global installation works fine otherwise, but I keep getting errors regarding temporary file creation. Perhaps that is fixed in the dev-master setup too.

I'll get back when I learn more.

**Comment by rask** [comment]
Okay, `dev-master` fixed the issues, and the temporary file error stemmed from invalid compression option.

What are the valid values for compression in `box.json`? According to `help build` `GZ` and `BZ2` are fine, but with `GZ` it fails.

Thanks!

**Comment by theofidry** [comment]
Weird, sorry can't look atm. What error do you get exactly?

**Comment by theofidry** [comment]
Hm after looking into this I found an issue. So the compression algorithm works fine on my end: the PHAR is properly compressed. However, `Phar::isCompressed()` or `box info myphar.phar` returns respectively `false` and `None`. This is due to Box using `Phar::compressFiles()` instead of `Phar::compress()`. In theory it does the same thing, in practice the second override the stub file. Box uses a custom stub file so this is a no go, and unfortunately because of that the result is a lie.

**Comment by theofidry** [comment]
Closed by #93
